charmcraft.yaml

```diff
@@ -90,11 +89,275 @@ config:
+      type: int
+
+    max-days-of-only-failed-jobs-before-connection-disable:
+      description: Number of consecuative days of only failed jobs before the connection is disabled.
+      type: int
+
+    max-failed-jobs-in-a-row-before-connection-disable:
+      description: Number of consecuative failed jobs before the connection is disabled.
+      type: int
+
+    ##### Worker config #####
+    max-spec-workers:
+      description: Maximum number of Spec workers each Airbyte Worker container can support. Defaults to 5.
+      default: 5
+      type: int
+
+    max-check-workers:
+      description: Maximum number of Check workers each Airbyte Worker container can support. Defaults to 5.
+      default: 5
+      type: int
+
+    max-sync-workers:
+      description: Maximum number of Sync workers each Airbyte Worker container can support. Defaults to 5.
+      default: 5
+      type: int
+
+    max-discover-workers:
+      description: Maximum number of Discover workers each Airbyte Worker container can support. Defaults to 5.
+      default: 5
+      type: int
+
+    ##### Data retention config #####
+    temporal-history-retention-in-days:
+      description: Retention period of the job history in Temporal, defaults to 30 days.
+      default: 30
+      type: int
+
+    ##### Kubernetes config #####
+    job-kube-tolerations:
+      description: |
+        Defines one or more Job pod tolerations. 
+        
+        Tolerations are separated by ';'. Each toleration contains k=v pairs mentioning some/all 
+        of key, effect, operator and value and separated by ','.
+      type: string
+
+    job-kube-node-selectors:
+      description: |
+        Defines one or more Job pod node selectors. 
+        
+        Each k=v pair is separated by a ','. For example: key1=value1,key2=value2. 
+        It is the pod node selectors of the sync job and the default pod node selectors
+        fallback for others jobs.
+      type: string
+
+    job-kube-annotations:
+      description: |
+        Defines one or more Job pod annotations.
+        
+        Each k=v pair is separated by a ','. For example: key1=value1,key2=value2.
+        It is the pod annotations of the sync job and the default pod annotations
+        fallback for others jobs.
+      type: string
+
+    job-kube-main-container-image-pull-policy:
+      description: Defines the Job pod connector image pull policy.
+      default: "IfNotPresent"
+      type: string
+
+    job-kube-main-container-image-pull-secret:
+      description: Defines the Job pod connector image pull secret. Useful when hosting private images.
+      type: string
+
+    job-kube-sidecar-container-image-pull-policy:
+      description: |
+        Defines the image pull policy on the sidecar containers in the Job pod. 
+        
+        Useful when there are cluster policies enforcing to always pull.
+      default: "IfNotPresent"
+      type: string
+
+    job-kube-socat-image:
```

Comment:
What are example values of these -image config values? A container registry URL?

Reply:
Yup, updated the docs to include examples.